src/python/grpcio_tests/tests/observability/_observability_test.py

```diff
@@ -0,0 +1,353 @@
+
+    def __init__(self, metrics: List[grpc_observability.StatsData],
+                 spans: List[grpc_observability.TracingData]):
+        self.span_collecter = spans
+        self.metric_collecter = metrics
+        self._server = None
+
+    def export_stats_data(
+            self, stats_data: List[grpc_observability.StatsData]) -> None:
+        self.metric_collecter.extend(stats_data)
+
+    def export_tracing_data(
+            self, tracing_data: List[grpc_observability.TracingData]) -> None:
+        self.span_collecter.extend(tracing_data)
+
+
+def handle_unary_unary(request, servicer_context):
+    return _RESPONSE
+
+
+def handle_unary_stream(request, servicer_context):
+    for _ in range(STREAM_LENGTH):
+        yield _RESPONSE
+
+
+def handle_stream_unary(request_iterator, servicer_context):
+    return _RESPONSE
+
+
+def handle_stream_stream(request_iterator, servicer_context):
+    for request in request_iterator:
+        yield _RESPONSE
+
+
+class _MethodHandler(grpc.RpcMethodHandler):
+
+    def __init__(self, request_streaming, response_streaming):
+        self.request_streaming = request_streaming
+        self.response_streaming = response_streaming
+        self.request_deserializer = None
+        self.response_serializer = None
+        self.unary_unary = None
+        self.unary_stream = None
+        self.stream_unary = None
+        self.stream_stream = None
+        if self.request_streaming and self.response_streaming:
+            self.stream_stream = lambda x, y: handle_stream_stream(x, y)
+        elif self.request_streaming:
+            self.stream_unary = lambda x, y: handle_stream_unary(x, y)
+        elif self.response_streaming:
+            self.unary_stream = lambda x, y: handle_unary_stream(x, y)
+        else:
+            self.unary_unary = lambda x, y: handle_unary_unary(x, y)
+
+
+class _GenericHandler(grpc.GenericRpcHandler):
+
+    def service(self, handler_call_details):
+        if handler_call_details.method == _UNARY_UNARY:
+            return _MethodHandler(False, False)
+        elif handler_call_details.method == _UNARY_STREAM:
+            return _MethodHandler(False, True)
+        elif handler_call_details.method == _STREAM_UNARY:
+            return _MethodHandler(True, False)
+        elif handler_call_details.method == _STREAM_STREAM:
+            return _MethodHandler(True, True)
+        else:
+            return None
+
+
+class ObservabilityTest(unittest.TestCase):
+
+    def setUp(self):
+        self.all_metric = []
+        self.all_span = []
+        self.test_exporter = TestExporter(self.all_metric, self.all_span)
+
+    def tearDown(self):
+        os.environ['GRPC_GCP_OBSERVABILITY_CONFIG'] = ''
+
+    def testRecordUnaryUnary(self):
+        os.environ[
+            'GRPC_GCP_OBSERVABILITY_CONFIG'] = _VALID_CONFIG_TRACING_STATS
+        with grpc_observability.GCPOpenCensusObservability() as o11y:
+            o11y.init(exporter=self.test_exporter)
```

Comment:
Is there any reason not to initialize with the context manager. That is, to call `init` as part of `grpc_observability.GCPOpenCensusObservability().__enter__`?

Reply:
I do remember I have some reasons to separate them, but I don't recall :(, I'll move it to `__enter__`.